src/arrayMethodJoin.js

```diff
@@ -4,8 +4,36 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(initialSeparator) {
+    let joinedArrString = '';
+    let separator = initialSeparator;
+    const newArr = [...this];
+
+    if (separator === undefined) {
+      separator = ',';
+    }
+
+    if (separator === null) {
+      separator = 'null';
+    }
+
+    for (let i = 0; i < newArr.length; i++) {
```

Comment:
Instead of modifying the original array, directly add the processed element to the result string.